contracts/contracts/link_token.cairo

```diff
@@ -0,0 +1,85 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.math import assert_not_zero, assert_le
+from starkware.cairo.common.uint256 import Uint256
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.bool import FALSE, TRUE
+from openzeppelin.token.erc20.library import ERC20
+from openzeppelin.token.erc20.presets.ERC20Upgradeable import (
+    initializer,
+    upgrade,
+    name,
+    symbol,
+    totalSupply,
+    decimals,
+    balanceOf,
+    allowance,
+    transfer,
+    transferFrom,
+    approve,
+    increaseAllowance,
+    decreaseAllowance,
+)
+from starkware.starknet.std_contracts.ERC20.permitted import (
+    permitted_initializer,
+    permitted_minter_only,
+    permittedMinter,
+)
+from contracts.ERC677.library import ERC677
+from contracts.ERC677.interfaces.IERC677Receiver import IERC677Receiver
+
+const NAME = 'ChainLink Token'
+const SYMBOL = 'LINK'
+
+@constructor
+func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
```

Comment:
@archseer as we agree, I will change this to be `initializer` the thing is that we already have a function name ìnitializer` in ERC20Upgradeable. Two possibility: - I can just use a different name like `link_initializer` 
- Or we use the `initializer` function from ERC20Upgradeable and then we will have to call  `permitted_initializer` to be able to use `permissionedMint` and  `permissionedBurn`

Reply:
After talking with @krebernisak, let's just use `ERC20` instead of `ERC20Upgradeable` and use a regular constructor

Reply:
Hmm actually, if we want to upgrade the token we do need to use `ERC20Upgradeable` since the bridge won't allow changing the token address: https://github.com/starkware-libs/starkgate-contracts/blob/c08863a1f08226c09f1d0748124192e848d73db9/src/starkware/starknet/apps/starkgate/cairo/token_bridge.cairo#L163-L167

Reply:
I would create a custom initializer:

```cairo
# This is the ERC20Upgradeable initializer but it also calls permitted_initializer
@external
func initializer{
        syscall_ptr: felt*,
        pedersen_ptr: HashBuiltin*,
        range_check_ptr
    }(
        name: felt,
        symbol: felt,
        decimals: felt,
        initial_supply: Uint256,
        recipient: felt,
        proxy_admin: felt
    ):
    ERC20.initializer(name, symbol, decimals)
    ERC20._mint(recipient, initial_supply)
    Proxy.initializer(proxy_admin)
    permitted_initializer()
    return ()
end
```